device/iot-device-client/src/main/java/com/microsoft/azure/sdk/iot/device/transport/amqps/AmqpsIotHubConnection.java

```diff
@@ -952,41 +952,41 @@ private void checkForNewlyUnregisteredMultiplexedClientsToStop()
             {
                 log.trace("Removing session handler for device {}", amqpsSessionHandler.getDeviceId());
                 this.sessionHandlers.remove(amqpsSessionHandler);
-            }
-            else
-            {
-                log.warn("Attempted to remove device session for device {} from multiplexed connection, but device was not currently registered.", configToUnregister.getDeviceId());
-                return;
```

Comment:
The issue here was that this code forgot to add this config to the set of "successfully" unregistered configs so that the end of the function could remove it from the queue of configs to unregister. It can't be removed at this point since that would be a concurrent modification exception since this function loops over that queue.